libia2/ia2.c

```diff
@@ -123,6 +124,49 @@ static bool in_loaded_segment(struct dl_phdr_info *info, Elf64_Addr address) {
   return false;
 }
 
+static bool in_extra_libraries(struct dl_phdr_info *info, const char *extra_libraries) {
+  if (!extra_libraries) {
+    return false;
+  }
+  char *library_name = basename(info->dlpi_name);
+  size_t library_name_len = strlen(library_name);
+  if (library_name_len == 0) {
+    return false;
+  }
+  size_t extra_libraries_len = strlen(extra_libraries);
+  size_t extra_libraries_pos = 0;
+  size_t library_name_pos = 0;
+  while (extra_libraries_pos < extra_libraries_len) {
+    // Compare this extra library with the current library
+    for (library_name_pos = 0; extra_libraries_pos < extra_libraries_len &&
+                               library_name_pos < library_name_len;
+         extra_libraries_pos++, library_name_pos++) {
+      if (extra_libraries[extra_libraries_pos] !=
```

Comment:
Shouldn't this use strcmp or am I missing something?

Reply:
This is doing per-char comparison, but strncmp would work instead, and is probably cleaner. I'll update this.

Reply:
Actually on second thought, this is simpler. I'll comment it better. We're actually comparing substrings to substrings, so using a `strncmp` would require `strchr` too.

Reply:
Oh right, I misread the arg type as `char **` and thought you were comparing pointers at first. I went through it again and it looks correct.